Source/WebKit/WebProcess/Plugins/PDF/WKAccessibilityPDFDocumentObject.mm

```diff
@@ -70,6 +85,69 @@ - (void)setPDFDocument:(RetainPtr<PDFDocument>)document
     _pdfDocument = document;
 }
 
+- (void)setParent:(NSObject *)parent
+{
+    _parent = parent;
+}
+
+- (PDFDocument *)document
+{
+    return _pdfDocument.get();
+}
+
+- (NSRect)convertFromPDFPageToScreenForAccessibility:(NSRect)rectInPageCoordinates pageIndex:(WebKit::PDFDocumentLayout::PageIndex)pageIndex
+{
+    if (RefPtr plugin = _pdfPlugin.get())
+        return plugin->convertFromPDFPageToScreenForAccessibility(rectInPageCoordinates, pageIndex);
+    return rectInPageCoordinates;
+}
+
+#if !PLATFORM(MAC)
+
+- (void)setAXElements
+{
+    if (!_pdfDocument) {
+        if (RefPtr plugin = _pdfPlugin.get())
+            _pdfDocument = plugin->pdfDocument();
+    }
+
+    auto pageCount = [_pdfDocument pageCount];
+    for (NSUInteger pageIndex = 0; pageIndex < pageCount; pageIndex ++) {
+        PDFPage *page = [_pdfDocument pageAtIndex:pageIndex];
+
+        if ([page respondsToSelector:@selector(accessibilityElementsWithPlugin:)])
+            [_axElements addObjectsFromArray:[page accessibilityElementsWithPlugin:self]];
+    }
+}
+
+- (id)accessibilityHitTest:(CGPoint)point
+{
+    if (!_pdfDocument) {
+        if (RefPtr plugin = _pdfPlugin.get())
+            _pdfDocument = plugin->pdfDocument();
+    }
+
+    if (RefPtr plugin = _pdfPlugin.get())
+        return plugin->accessibilityHitTestInPage(point);
+    return [_pdfDocument accessibilityHitTest:point];
+}
+
+- (NSArray *)accessibilityElements
+{
+    if ([_axElements count])
+        [self setAXElements];
```

Comment:
Did you mean to call `setAXElements` if `_axElements` is already non-empty? I thought it would read more like
```suggestion
    if (![_axElements count])
        [self setAXElements];
```